lib/mdx.ml

```diff
@@ -56,16 +56,27 @@ let filter_section re (t : t) =
   | l -> Some l
 
 let parse l =
-  List.map
-    (function
-      | `Text t -> Text t | `Section s -> Section s | `Block b -> Block b)
-    l
+  let results =
+    List.map
+      (function
+        | `Text t -> Ok (Text t)
+        | `Section s -> Ok (Section s)
+        | `Block rb -> Block.from_raw rb >>= fun b -> Ok (Block b))
+      l
+  in
+  let errors =
+    Util.List.concat_map (function Ok _ -> [] | Error l -> l) results
+  in
+  let ok =
+    List.filter_map (function Ok x -> Some x | Error _ -> None) results
+  in
```

Comment:
Since `concat_map` has to be supplied by us for compatibility reasons, wouldn't it be better to implement `Base.List.partition_result`/`Base.List.partition_map`? That saves one iteration and is clearer on what it does.

Reply:
Yeah good point! I'll see how I can improve it!